test-clients/src/main/java/com/hedera/services/bdd/suites/contract/WarmUpAccountOrStorageSuite.java

```diff
@@ -0,0 +1,114 @@
+package com.hedera.services.bdd.suites.contract;
+
+import com.hedera.services.bdd.spec.HapiApiSpec;
+import com.hedera.services.bdd.spec.infrastructure.meta.ContractResources;
+import com.hedera.services.bdd.spec.utilops.CustomSpecAssert;
+import com.hedera.services.bdd.suites.HapiApiSuite;
+import com.hedera.services.utils.EntityIdUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.units.bigints.UInt256;
+
+import java.util.List;
+
+import static com.hedera.services.bdd.spec.HapiApiSpec.defaultHapiSpec;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getTxnRecord;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.contractCall;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.contractCreate;
+import static com.hedera.services.bdd.spec.transactions.TxnVerbs.fileCreate;
+import static com.hedera.services.bdd.spec.utilops.UtilVerbs.withOpContext;
+import static com.hederahashgraph.api.proto.java.ResponseCodeEnum.SUCCESS;
+
+public class WarmUpAccountOrStorageSuite extends HapiApiSuite {
+
+	private static final Logger log = LogManager.getLogger(WarmUpAccountOrStorageSuite.class);
+
+	public static void main(String... args) {
+		new WarmUpAccountOrStorageSuite().runSuiteSync();
+	}
+
+	@Override
+	protected Logger getResultsLogger() {
+		return log;
+	}
+
+	@Override
+	protected List<HapiApiSpec> getSpecsInSuite() {
+		return List.of(
+				contractCreateWithAccessedKeys()
```

Comment:
I think that we should implement the following e2e test cases:
- When providing one contract in the access list, the gas cost is `-100`
- Providing keys `> bytes32` throws during precheck with new response code `INVALID_STORAGE_ACCESS_KEY`
- When providing one contract + 2 storage keys, the gas cost is `-500`

Suggestion for test names:
- `accountWarmUpReducesGasCost`
- `storageWramUpReducesGasCost`
- `rejectsInvalidStorageAccessKey`